src/iuse.cpp

```diff
@@ -3847,7 +3847,7 @@ int iuse::set_trap(player *p, item *it, bool, const tripoint& )
     }
 
     if (buried) {
-        if (!p->has_amount("shovel", 1) && !p->has_amount("e_tool", 1)) {
+        if (!p->has_items_with_quality( "DIG", 2, 1 )) {
             p->add_msg_if_player(m_info, _("You need a shovel."));
```

Comment:
At one pint, those messages (including the other for the crowbar) should be rephrased to contain the actual requirement. As in "You need a digging tool" or similar to refer to the quality instead of an actual item.